sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+static az_result build_confirmed_reported_property(
+    az_json_builder* json_builder,
+    az_span property_name,
+    double property_val,
+    int32_t ac,
+    int32_t av,
+    az_span ad)
+{
+  AZ_RETURN_IF_FAILED(az_json_builder_init(json_builder, AZ_SPAN_FROM_BUFFER(reported_property_payload), NULL));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_begin_object(json_builder));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, property_name));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_begin_object(json_builder));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, AZ_SPAN_FROM_STR("value")));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, (int32_t)property_val));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, AZ_SPAN_FROM_STR("ac")));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, ac));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, AZ_SPAN_FROM_STR("av")));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, av));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, AZ_SPAN_FROM_STR("ad")));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_string(json_builder, ad));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_end_object(json_builder));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_end_object(json_builder));
+
+  return AZ_OK;
+}
+
+static az_result build_reported_property(
+    az_json_builder* json_builder,
+    az_span property_name,
+    double property_val)
+{
+  AZ_RETURN_IF_FAILED(az_json_builder_init(json_builder, AZ_SPAN_FROM_BUFFER(reported_property_payload), NULL));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_begin_object(json_builder));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, property_name));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, (int32_t)property_val));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_end_object(json_builder));
+
+  return AZ_OK;
+}
+// Send the twin reported property to the service
+static int send_reported_temperature_property(double temp_value, int32_t version)
+{
+  int rc;
+  printf("Sending reported property\n");
+
+  // Get the topic used to send a reported property update
+  az_span request_id_span = get_request_id();
+  if (az_failed(
+          rc = az_iot_hub_client_twin_patch_get_publish_topic(
+              &client,
+              request_id_span,
+              reported_property_topic,
+              sizeof(reported_property_topic),
+              NULL)))
+  {
+    printf("Unable to get twin document publish topic, return code %d\n", rc);
+    return rc;
+  }
+
+  // Twin reported properties must be in JSON format. The payload is constructed here.
+  az_json_builder json_builder;
+  if (version < 0)
+  {
+    if (az_failed(
+            rc = build_reported_property(
+                &json_builder, AZ_SPAN_FROM_STR("maxTempSinceLastReboot"), temp_value)))
```

Comment:
>maxTempSinceLastReboot [](start = 49, length = 22)

maxTempSinceLastreboot is hard-coded here should be at top of file.

Reply:
We might want to come to a consensus about this since the team has differing opinions about these. I think I had it up top originally